File: app/src/main/java/edu/gatech/cs2340/spacetraderredux/Views/SystemInfoActivity.java
package edu.gatech.cs2340.spacetraderredux.Views;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.DefaultItemAnimator;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import java.util.LinkedList;

import edu.gatech.cs2340.spacetraderredux.Model.ResourcesViewAdapter;
import edu.gatech.cs2340.spacetraderredux.Model.TempTrade;
import edu.gatech.cs2340.spacetraderredux.Model.TradeViewAdapter;
import edu.gatech.cs2340.spacetraderredux.R;

public class SystemInfoActivity extends AppCompatActivity {
    public ResourcesViewAdapter adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_system_info);
        TextView tabText = (TextView) findViewById(R.id.labelTabName);
        tabText.setText("System Info");

        LinkedList<TempTrade> trades = new LinkedList<TempTrade>();
        trades.add(new TempTrade());
        trades.add(new TempTrade());
        trades.add(new TempTrade());

        RecyclerView recyclerView = (RecyclerView) findViewById(R.id.recyclerView);

        adapter = new ResourcesViewAdapter(trades);
        RecyclerView.LayoutManager mLayoutManager = new GridLayoutManager(getApplicationContext(), 2);
        recyclerView.setLayoutManager(mLayoutManager);
        recyclerView.setItemAnimator(new DefaultItemAnimator());
        recyclerView.setAdapter(adapter);
        adapter.notifyDataSetChanged();
    }

    public void systemInfoClick(View view) {
        Intent activityChangeIntent = new Intent(SystemInfoActivity.this, SystemInfoActivity.class);
        SystemInfoActivity.this.startActivity(activityChangeIntent);
    }

    public void cargoClick(View view) {
        Intent activityChangeIntent = new Intent(SystemInfoActivity.this, CargoActivity.class);
        SystemInfoActivity.this.startActivity(activityChangeIntent);
    }

    public void tradeableClick(View view) {
        Intent activityChangeIntent = new Intent(SystemInfoActivity.this, Trade.class);
        SystemInfoActivity.this.startActivity(activityChangeIntent);
    }
}

File: app/src/main/java/edu/gatech/cs2340/spacetraderredux/ui/tradespec/SystemInfoActivity.kt
package edu.gatech.cs2340.spacetraderredux.ui.tradespec

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.DefaultItemAnimator
import android.support.v7.widget.GridLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.TextView

import java.util.LinkedList

import edu.gatech.cs2340.spacetraderredux.R

class SystemInfoActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_system_info)
        val tabText = findViewById<View>(R.id.labelTabName) as TextView
        tabText.text = "System Info"

        val trades = LinkedList<TempTrade>()
        trades.add(TempTrade())
        trades.add(TempTrade())
        trades.add(TempTrade())

        val recyclerView = findViewById<View>(R.id.recyclerView) as RecyclerView

        var adapter = ResourcesViewAdapter(trades)
        val mLayoutManager = GridLayoutManager(applicationContext, 2)
        recyclerView.layoutManager = mLayoutManager
        recyclerView.itemAnimator = DefaultItemAnimator()
        recyclerView.adapter = adapter
        adapter.notifyDataSetChanged()
    }

    fun systemInfoClick(view: View) {
        val activityChangeIntent = Intent(this@SystemInfoActivity, SystemInfoActivity::class.java)
        this@SystemInfoActivity.startActivity(activityChangeIntent)
    }

    fun cargoClick(view: View) {
        val activityChangeIntent = Intent(this@SystemInfoActivity, CargoActivity::class.java)
        this@SystemInfoActivity.startActivity(activityChangeIntent)
    }

    fun tradeableClick(view: View) {
        val activityChangeIntent = Intent(this@SystemInfoActivity, Trade::class.java)
        this@SystemInfoActivity.startActivity(activityChangeIntent)
    }
}
